### PR TITLE
Option value scalars are now always strings but include type information.

### PR DESCRIPTION
**Pull request by JakeWharton**
This also changes rendering the schema from implicitly being behind `toString()` to an explicit `toSchema()`.

### CODE REVIEW

**Comment by JakeWharton** [comment]
@swankjesse

File: src/test/java/com/squareup/protoparser/ProtoParserTest.java [hunk 1]
```diff
@@ -919,6 +926,148 @@
...
+                .addOption(OptionElement.create("default", Kind.NUMBER, "123.456e7"))
+                .build())
+            .addField(FieldElement.builder()
+                .label(OPTIONAL)
+                .type(ScalarType.DOUBLE)
+                .name("default_double")
+                .tag(413)
+                .addOption(OptionElement.create("default", Kind.NUMBER, "123.456e78"))
+                .build())
+            //.addField(FieldElement.builder()
```
Comment (swankjesse) [inline src/test/java/com/squareup/protoparser/ProtoParserTest.java]: curious

Comment (swankjesse) [inline src/test/java/com/squareup/protoparser/ProtoParserTest.java]: ahhh, escaping

Comment (JakeWharton) [inline src/test/java/com/squareup/protoparser/ProtoParserTest.java]: Yep. I'll get there soon!

File: src/test/java/com/squareup/protoparser/ProtoParserTest.java [hunk 2]
```diff
@@ -919,6 +926,148 @@
...
+            //    .label(OPTIONAL)
+            //    .type(ScalarType.STRING)
+            //    .name("default_string")
+            //    .tag(414)
+            //    .addOption(OptionElement.create("default", Kind.STRING,
+            //        "çok\\a\\b\\f\\n\\r\\t\\v\\1\\01\\001\\17\\017\\176\\x1\\x01\\x11\\X1\\X01\\X11güzel"))
```
Comment (swankjesse) [inline src/test/java/com/squareup/protoparser/ProtoParserTest.java]: I think you want the equals() for `Kind.STRING` to be equal after conversion. Which means you probably need to decode it when you create the option.

**Pull request merged**